e2e: e2e_node: refactor getCurrentKubeletConfig

ffromani opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Refactoring to de-dup the getCurrentKubeletConfig helper which is used by both e2e and e2e_node tests. This patch originated because we want to add more topology manager e2e tests, and the function started to proliferate too much. See https://github.com/kubernetes/kubernetes/pull/88107

**Which issue(s) this PR fixes**:
Related to https://github.com/kubernetes/kubernetes/issues/83481

**Special notes for your reviewer**:
This patch works for me running e2e_node tests locally, but I don't have windows machine available.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

ffromani committed:
e2e: e2e_node: refactor getCurrentKubeletConfig

this patch moves the helper getCurrentKubeletConfig function,
used in both e2e and e2e_node tests and previously duplicated,
in the common framework.

There are no intended changes in behaviour.

Signed-off-by: Francesco Romani <fromani@redhat.com>

fromanirh commented:
/sig testing

fromanirh commented:
/assign cblecker

oomichi commented on test/e2e/framework/kubeletconfig.go:
```diff
@@ -0,0 +1,131 @@
+/*
```
Could you implement this as `test/e2e/framework/kubelet/config.go`?

If implementing this as `test/e2e/framework/kubeletconfig.go`, this file is defined as core part of e2e test framework.
This file imports `e2ekubectl` which is defined as subpackage of e2e test framework, and that means invalid dependency as https://github.com/kubernetes/kubernetes/issues/81245
By implementing as ` test/e2e/framework/kubelet/config.go` we can avoid such thing.

oomichi replied:
/cc @oomichi

fromanirh replied:
Sure thing @oomichi , on it

ffromani committed:
e2e: getCurrentKubeletConfig: move in subpkg

Address review comments and move the helper function
in the `framework/kubelet` package to avoid circular deps
(see https://github.com/kubernetes/kubernetes/issues/81245)

Signed-off-by: Francesco Romani <fromani@redhat.com>

fromanirh commented:
not sure who can review/approve from windows side @michmike and @PatrickLang (according to the OWNERS file) could you please help?

oomichi commented:
Thanks for updating.

/ok-to-test

oomichi commented:
Need to run `./hack/update-bazel.sh` for passing pull-kubernetes-verify job.

ffromani committed:
update bazel configuration

Signed-off-by: Francesco Romani <fromani@redhat.com>

fromanirh commented:
/sig node

PatrickLang commented:
/assign
Will test on Windows

oomichi commented:
Thanks for updating.

/lgtm

derekwaynecarr commented:
/approve

/hold

@PatrickLang when you have a chance to confirm, please remove the hold.

SataQiu reviewed (commented):
/approve
Nice work!
defer @PatrickLang to confirm and remove the hold

PatrickLang commented:
/test pull-kubernetes-e2e-aks-engine-azure-windows

I'm not sure the state of that prow job, so I'm scheduling my own local run too

PatrickLang commented:
/hold cancel

Seems ok - the moved code works. The test case still fails but I'll file a separate issue for that. It's not on testgrid at the moment because we're skipping [slow][serial].

PatrickLang commented:
Filed #88481 for broken Windows test
